plugins/jvm/ts/discover.ts

```diff
@@ -4,28 +4,31 @@ namespace JVM {
 
   _module.controller("JVM.DiscoveryController", ["$scope", "localStorage", "jolokia", ($scope, localStorage, jolokia) => {
 
-    $scope.discovering = true;
-    $scope.agents = <any>undefined;
+    $scope.jvmDiscovery = {
+      discovering : true,
+      responseJson : undefined,
+      agents : <any>undefined,
+      selectedAgent: undefined,
+      filter: undefined
+    };
 
-    $scope.$watch('agents', (newValue, oldValue) => {
+    $scope.$watch('jvmDiscovery.agents', (newValue, oldValue) => {
       if (newValue !== oldValue) {
-        $scope.selectedAgent = _.find($scope.agents, a => a['selected']);
+        $scope.jvmDiscovery.selectedAgent = _.find($scope.jvmDiscovery.agents, a => a['selected']);
       }
     }, true);
 
     $scope.closePopover = ($event) => {
-      (<any>$)($event.currentTarget).parents('.popover').prev().popover('hide');
+      (<any>$)($event.currentTarget).parents('.popover').hide();
     };
 
     function getMoreJvmDetails(agents) {
       for (let key in agents) {
         const agent = agents[key];
         if (agent.url && !agent.secured) {
-          const dedicatedJolokia = createJolokia(Core.useProxyIfExternal(agent.url), agent.username, agent.password);
+          const dedicatedJolokia = createJolokia(agent.url, agent.username, agent.password);
           agent.startTime = dedicatedJolokia.getAttribute('java.lang:type=Runtime', 'StartTime');
-          if (!$scope.hasName(agent)) {//only look for command if agent vm is not known
-            agent.command = dedicatedJolokia.getAttribute('java.lang:type=Runtime', 'SystemProperties', 'sun.java.command');
-          }
+          agent.command = dedicatedJolokia.getAttribute('java.lang:type=Runtime', 'SystemProperties', 'sun.java.command');
```

Comment:
Why is `$scope.hasName(agent)` check removed?

Reply:
Hi, I found an inconsistency in the UI where the java command would not be printed when mocking server detail.
The if clause is "arbitrary" and the comment doesn't help understanding the logic of the author behind this. Maybe you can shed light  on why shouldn't we look for the java command when agent VM is known? 

From a UI point of view, this leads to element in the list not displaying the same information which looks like a defect at first glance.

Reply:
@tadayosi  see above comment.